🐛 Make release expecting to build binaries

The original `Makefile` from CAPI expects to build `clusterctl` via `make release-binaries` and in CAPN we don't have these. We need to remove that from the `make release` target.

```
# Build binaries first.
/Applications/Xcode.app/Contents/Developer/usr/bin/make release-binaries
make[1]: *** No rule to make target `release-binaries'.  Stop.
make: *** [release] Error 2
error: current-context must exist in order to minify

```

--- a/Makefile
+++ b/Makefile
@@ -324,8 +324,6 @@ release: clean-release ## Builds and push container images using the latest git 
 	@if [ -z "${RELEASE_TAG}" ]; then echo "RELEASE_TAG is not set"; exit 1; fi
 	@if ! [ -z "$$(git status --porcelain)" ]; then echo "Your local git repository contains uncommitted changes, use git clean before proceeding."; exit 1; fi
 	git checkout "${RELEASE_TAG}"
-	# Build binaries first.
-	$(MAKE) release-binaries
 	# Set the core manifest image to the production bucket.
 	$(MAKE) set-manifest-image \
 		MANIFEST_IMG=$(PROD_REGISTRY)/$(IMAGE_NAME) MANIFEST_TAG=$(RELEASE_TAG) \

